src/microservices/Microservices.DicomTagReader/Execution/TagReaderBase.cs

```diff
@@ -85,11 +86,13 @@ public void ReadTags(IMessageHeader header, AccessionDirectoryMessage message)
             if (!dirPath.StartsWith(_filesystemRoot, StringComparison.CurrentCultureIgnoreCase))
                 throw new ApplicationException("Directory " + dirPath + " is not below the given FileSystemRoot (" +
                                                _filesystemRoot + ")");
-
             long beginEnumerate = _stopwatch.ElapsedTicks;
             string[] dicomFilePaths = _fs.Directory.EnumerateFiles(dirPath, _searchPattern).ToArray();
+            string[] zipFilePaths =  _fs.Directory.EnumerateFiles(dirPath).Where(f=>Path.GetExtension(f) == ".zip" || Path.GetExtension(f) == ".tar").ToArray();
```

Comment:
On Linux, I think we're more likely to see gzipped archives rather than bare tar files -- so *.tar.gz and *.tgz. We should be able to use GZipStream to handle this case though.

Reply:
Do we want to move the IsZipFile into a helper class? so the definition is standard and if needed wrap the openread for different file types?

Reply:
I think that would make sense, yes. We're also trusting the file extension rather than checking the file contents before trying to read -- I assume the read will fail with a sensible error though if that happens.

Reply:
Ok I have moved the extension evaluation to ZipHelper https://github.com/SMI/SmiServices/pull/327/commits/636aff7544c3643dad700e58eb1e75e3e85621c7 

If we need to enumerate different archive types and serve dicomdatasets out of them then I would rather move that class up to Rdmp.Dicom or DicomTypeTranslation since its quite a big piece of functionality and theres already archive handling stuff further up e.g. [ZipPool](https://github.com/HicServices/RdmpDicom/blob/52e10f9d323f63e2998c8794d0afa5587a0801de/Rdmp.Dicom/Extraction/FoDicomBased/ZipPool.cs) which handles random access into zip files.